### PR TITLE
Utils: kvlds-dump, kvlds-undump

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This is a revised version of #106.  I've attached a diff of the `kvlds-dump/` and the `kvlds-undump/` directories between these branches.

You can generate this locally:
```
$ git diff origin/utils..origin/utils-dump -- kvlds-dump/ kvlds-undump/
```

or, if you've switched to this branch:
```
$ git diff origin/utils..HEAD -- kvlds-dump/ kvlds-undump/
```

If you don't specify certain directories (with `-- dirname(s)`), you'll get diffs in the entire file tree, which includes a bunch of unrelated build-system and python changes.


[diff.txt](https://github.com/Tarsnap/kivaloo/files/3224958/diff.txt)

**Comment by gperciva** [comment]
Oh, we discussed `events_shutdown()`.  Changes to that won't be coming very soon, so I don't think it's worth postponing these utilities for that.

File: kvlds-undump/main.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+	/* Print statistics if appropriate. */
+	if (opt_v) {
+		fprintf(stderr,
+		    "Stored %" PRIu64 " key-value pairs in %f seconds.",
+		    C.N, timeval_diff(en, st));
+	}
+
+	/* Close the directory (if applicable). */
+	if (opt_fs) {
```
Comment (cperciva) [inline kvlds-undump/main.c]: Style: Unnecessary braces.

Comment (gperciva) [reply kvlds-undump/main.c]: Solved: added stuff to make the braces not unnecessary.

**Comment by cperciva** [comment]
LGTM

**Comment by gperciva** [comment]
Update PR to rebase on master.  Ready for merge.

At the last moment, I fixed the commit message of the `Add test...` commit, which still mentioned building a dummy program due to the build system.

**Comment by cperciva** [comment]
Please fix conflicts.

**Pull request merged**